cmd/util/cmd/execution-state-extract/cmd.go

```diff
@@ -34,37 +35,49 @@ func init() {
 		"Directory to write new Execution State to")
 	_ = Cmd.MarkFlagRequired("output-dir")
 
+	Cmd.Flags().StringVar(&flagStateCommitment, "state-commitment", "",
+		"state commitment (hex-encoded, 64 characters)")
+
 	Cmd.Flags().StringVar(&flagBlockHash, "block-hash", "",
 		"Block hash (hex-encoded, 64 characters)")
-	_ = Cmd.MarkFlagRequired("block-hash")
 
 	Cmd.Flags().StringVar(&flagDatadir, "datadir", "",
 		"directory that stores the protocol state")
-	_ = Cmd.MarkFlagRequired("datadir")
 }
 
 func run(*cobra.Command, []string) {
+	var stateCommitment []byte
 
-	blockID, err := flow.HexStringToIdentifier(flagBlockHash)
-	if err != nil {
-		log.Fatal().Err(err).Msg("malformed block hash")
-	}
+	if len(flagBlockHash) > 0 {
+		blockID, err := flow.HexStringToIdentifier(flagBlockHash)
+		if err != nil {
+			log.Fatal().Err(err).Msg("malformed block hash")
+		}
 
-	db := common.InitStorage(flagDatadir)
-	defer db.Close()
+		db := common.InitStorage(flagDatadir)
+		defer db.Close()
 
-	cache := &metrics.NoopCollector{}
-	commits := badger.NewCommits(cache, db)
+		cache := &metrics.NoopCollector{}
+		commits := badger.NewCommits(cache, db)
 
-	stateCommitment, err := getStateCommitment(commits, blockID)
-	if err != nil {
-		log.Fatal().Err(err).Msg("cannot get state commitment for block")
+		stateCommitment, err = getStateCommitment(commits, blockID)
+		if err != nil {
+			log.Fatal().Err(err).Msg("cannot get state commitment for block")
+		}
+	}
+
+	if len(flagStateCommitment) > 0 {
```

Comment:
I see that if both flags are entered, priority is given to `flagBlockHash`. Shouldn't we not allow having both flags to make it clearer?

Reply:
Yup, I'll add the check and update the description.

Reply:
I added at c32ce98